project-addons/flask_middleware_connector/wizard/middleware_wizard.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Visiotech All Rights Reserved
+#    $Jesus Garcia Manzanas <jgmanzanas@visiotechsecurity.com>$
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, _
+from openerp.addons.connector.session import ConnectorSession
+from ..events.partner_events import export_partner, update_partner
+from ..events.product_events import update_product, export_product
+from ..events.rma_events import export_rma, export_rmaproduct, update_rma, update_rmaproduct
+from ..events.invoice_events import export_invoice, update_invoice
+from ..events.picking_events import export_picking, update_picking, export_pickingproduct, update_pickingproduct
+
+
+class MiddlewareBackend(models.Model):
+    _name = 'middleware.backend.export'
+
+    type_export = fields.Selection(
+        selection=[
+            ('partner', 'Partner'),
+            ('invoices', 'Invoices'),
+            ('pickings', 'Pickings'),
+            ('rmas', 'RMAs'),
+            ('products', 'Products')
+        ],
+        string='Export type',
+        required=True,
+    )
+    mode_export = fields.Selection(
+        selection=[
+            ('export', 'Export'),
+            ('update', 'Update')
+        ],
+        string='Export mode',
+        required=True,
+    )
+
+    @api.multi
+    def do_export(self):
+        session = ConnectorSession(self.env.cr, self.env.uid,
+                                   context=self.env.context)
+        if self.type_export == 'partner':
+            partner_obj = self.env['res.partner']
+            partner_ids = partner_obj.search([('is_company', '=', True),
+                                              ('web', '=', True),
+                                              ('customer', '=', True)])
+            contact_ids = partner_obj.search([('parent_id', 'in', partner_ids.ids),
```

Comment:
esto sería mejor: ('id', 'child_of', partner_ids.ids)

Reply:
sería más bien: ('id', 'child_of', partner_ids.ids),('id','not in', partner_ids.ids)
Así incluye todos los hijos de las empresas buscadas arriba pero no las empresas buscadas arriba, sino se estaría volviendo a exportar.